### ISSUE TITLE
Provider "registry.terraform.io/hashicorp/kubernetes" produced an invalid plan for kubernetes_secret causing terraform to crash

### ISSUE THREAD

**Issue by johnmehringer** [hashicorp/terraform-provider-kubernetes#2071]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform version: v1.4.4
Kubernetes provider version: v2.19.0
Kubernetes version: v1.25.4+k3s1 (colima)
```

### Affected Resource(s)
kubernetes_secret

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```
https://www.icloud.com/iclouddrive/03e8uEnjtGr6dcK0uqspY3Pyw#terraform-crash-example

### Debug Output

https://gist.github.com/johnmehringer/bc119d5755058c70adc14ec3152bd75f

### Panic Output

https://gist.github.com/johnmehringer/f0ac4076cd651b4a420fafd745df5d02

### Steps to Reproduce
1. `tar xvzf terraform-crash-example.tar.gz`
2. `cd terraform-crash-example/deployments/colima`
3. `terraform init`
4. `terraform apply`
5. `terraform apply`

### Expected Behavior
What should have happened?
The second `terraform apply` (step 5) should exit cleanly with no changes to be made.

### Actual Behavior
What actually happened?
terraform crashed

### Important Factoids

I am running my kubernetes cluster on macOS using colima

### References

- GH-1234

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by sheneska** [comment]
Hi @johnmehringer, Thanks for opening this issue, could you paste the `terraform configuration code` in the description please if it is not too big? 
Additionally, based on the stack trace report it looks like this could be a terraform issue, I would suggest you reopening this in the terraform provider.

**Comment by johnmehringer** [comment]
Hi @sheneska, I uploaded a tar.gz file with the terraform configuration code I used at the iCloud link which was provided in the bug report. Here it is for reference.

https://www.icloud.com/iclouddrive/03e8uEnjtGr6dcK0uqspY3Pyw#terraform-crash-example

Are you able to access it? If not, let me know and I'll try another cloud storage service. Thanks!

**Comment by johnmehringer** [comment]
It looks like this issue has been fixed in the terraform 1.4.5 release. I am no longer able to reproduce it after upgrading from 1.4.4.
